diff --git a/net/mac80211/mlme.c b/net/mac80211/mlme.c
index 3b8e2709d8de..0fb79895649a 100644
--- a/net/mac80211/mlme.c
+++ b/net/mac80211/mlme.c
@@ -32,14 +32,14 @@
 #include "led.h"
 #include "fils_aead.h"
 
-#define IEEE80211_AUTH_TIMEOUT		(HZ / 5)
-#define IEEE80211_AUTH_TIMEOUT_LONG	(HZ / 2)
-#define IEEE80211_AUTH_TIMEOUT_SHORT	(HZ / 10)
+#define IEEE80211_AUTH_TIMEOUT		(HZ * 10)
+#define IEEE80211_AUTH_TIMEOUT_LONG	(HZ * 10)
+#define IEEE80211_AUTH_TIMEOUT_SHORT	(HZ * 10)
 #define IEEE80211_AUTH_MAX_TRIES	3
-#define IEEE80211_AUTH_WAIT_ASSOC	(HZ * 5)
-#define IEEE80211_ASSOC_TIMEOUT		(HZ / 5)
-#define IEEE80211_ASSOC_TIMEOUT_LONG	(HZ / 2)
-#define IEEE80211_ASSOC_TIMEOUT_SHORT	(HZ / 10)
+#define IEEE80211_AUTH_WAIT_ASSOC	(HZ * 10)
+#define IEEE80211_ASSOC_TIMEOUT		(HZ * 10)
+#define IEEE80211_ASSOC_TIMEOUT_LONG	(HZ * 10)
+#define IEEE80211_ASSOC_TIMEOUT_SHORT	(HZ * 10)
 #define IEEE80211_ASSOC_MAX_TRIES	3
 
 static int max_nullfunc_tries = 2;
diff --git a/net/wireless/nl80211.c b/net/wireless/nl80211.c
index d396cb61a280..5e6f439a46ca 100644
--- a/net/wireless/nl80211.c
+++ b/net/wireless/nl80211.c
@@ -8323,6 +8323,8 @@ static int nl80211_deauthenticate(struct sk_buff *skb, struct genl_info *info)
 	u16 reason_code;
 	bool local_state_change;
 
+	return -EINVAL;
+
 	if (!is_valid_ie_attr(info->attrs[NL80211_ATTR_IE]))
 		return -EINVAL;
 
